tests/core/snapd-failover/task.yaml

```diff
@@ -1,64 +1,163 @@
+                exit 1
+            fi
+
+            echo "Verify that snapd.failure was activated when we tried to install a broken snapd"
+            #shellcheck disable=SC2165
+            for _ in $(seq 60); do
+                # get the number of times that snapd.failure was started
+                started_after="$(get_journalctl_log -u snapd.failure | grep -c 'Started Failure handling of the snapd snap.' || true)"
+                if [ "$started_after" -gt "$started_before" ] ; then
+                    break
+                fi
+                sleep 1
+            done
+
+            # just because snapd.failure.service is active doesn't mean that we are 
+            # fully ready; we should wait until the snap command shows up again
+            echo "And verify that snap commands still work and snapd is reverted"
+            retry-tool -n 30 --wait 1 bash -c "snap list | MATCH '^snapd .* $current .*'"
+
+            echo "Verify we got the expected error message"
+            snap change --last=install|MATCH "there was a snapd rollback across the restart"
+
+            echo "restart snapd and ensure we can still talk to it"
+            systemctl restart snapd.socket snapd.service
+
+            # we should still have the snapd snap and it should have the revision we had
+            # before the broken one was installed
+            snap list | MATCH "^snapd .* $current .*"
+        done
+
+        # we cannot undo the core -> snapd installation we did in the prepare 
+        # section during restore, because we can't reboot during restore, so instead
+        # do that restoration here
+        if is_core16_system; then
+            echo "Manually uninstall the snapd snap on UC16"
+            systemctl stop snapd.service snapd.socket snapd.autoimport.service snapd.snap-repair.service snapd.snap-repair.timer
+            umount "/snap/snapd/$(readlink /snap/snapd/current)"
+
+            rm -f /etc/systemd/system/usr-lib-snapd.mount
+            rm -f /etc/systemd/system/snap-snapd-*.mount
+            rm -f /etc/systemd/system/snapd.{service,timer,socket}
+            rm -f /etc/systemd/system/snapd.*.{service,timer,socket}
+            rm -f /etc/systemd/system/*.wants/snapd.*.{service,timer,socket}
+            systemctl daemon-reload
+            # this will have the "snapd" snap /usr/lib/snapd bind mounted
+            umount --lazy /usr/lib/snapd
+            systemctl start snapd.service snapd.socket
+            systemctl status snapd|MATCH " /usr/lib/snapd/snapd"
+
+            echo "Reboot to get a clean system again"
+            REBOOT
+        fi
+    else
+        # technically this check is unnecessary because we only reboot during
+        # the test's execution on uc16, but just be extra safe
+        if is_core16_system; then
+            # now remove the snapd snap since we booted with the core snap
+            snap list
+            snap remove snapd
```

Comment:
Can you add a comment that this only succeeds because the snapd snap is now in `broken` state.

Reply:
sure